packages/trader/src/Modules/Trading/Containers/__tests__/trade.spec.tsx

```diff
@@ -0,0 +1,171 @@
+    ...jest.requireActual('@deriv/shared'),
+    isDesktop: jest.fn(() => true),
+    isMobile: jest.fn(() => false),
+    TRADE_TYPES: jest.fn(),
+}));
+
+type TMockTrader = {
+    rootStore: TCoreStores;
+};
+
+const MockTrader = ({ rootStore }: TMockTrader) => (
+    <TraderProviders store={rootStore}>
+        <Trader />
+    </TraderProviders>
+);
+
+describe('Trader', () => {
+    let rootStore: TCoreStores;
+    beforeEach(() => {
+        rootStore = mockStore({
+            ui: {
+                notification_messages_ui: 'div',
+                is_mobile: false,
+            },
+            modules: {
+                trade: {
+                    contract_type: 'CALL',
+                    form_components: [],
+                    getFirstOpenMarket: jest.fn(),
+                    has_barrier: false,
+                    is_accumulator: false,
+                    is_chart_loading: false,
+                    is_market_closed: false,
+                    is_synthetics_available: false,
+                    is_synthetics_trading_market_available: false,
+                    is_trade_enabled: true,
+                    is_trade_params_expanded: false,
+                    is_turbos: false,
+                    is_vanilla: false,
+                    onMount: jest.fn(),
+                    onUnmount: jest.fn(),
+                    prepareTradeStore: jest.fn(),
+                    setIsDigitsWidgetActive: jest.fn(),
+                    setMobileDigitView: jest.fn(),
+                    should_show_active_symbols_loading: false,
+                    show_digits_stats: true,
+                    symbol: 'symbol',
+                },
+            },
+        });
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('renders the trader component in desktop view', () => {
+        render(<MockTrader rootStore={rootStore} />);
+        expect(screen.getByText('PositionsDrawer')).toBeInTheDocument();
+        expect(screen.queryByTestId('dt_swipeable')).not.toBeInTheDocument();
+        expect(screen.getByText('TradeChart')).toBeInTheDocument();
+        expect(screen.getByText('FormLayout')).toBeInTheDocument();
+    });
+
+    it('renders the trader component in mobile view', () => {
+        rootStore.ui.is_mobile = true;
+        (isDesktop as jest.Mock).mockReturnValue(false);
+        (isMobile as jest.Mock).mockReturnValue(true);
+        render(<MockTrader rootStore={rootStore} />);
+        expect(screen.queryByText('PositionsDrawer')).not.toBeInTheDocument();
+        expect(screen.getAllByTestId('dt_swipeable')).not.toHaveLength(0);
+        expect(screen.getByText('TradeChart')).toBeInTheDocument();
+        expect(screen.getByText('FormLayout')).toBeInTheDocument();
+    });
+
+    it('should show market is closed overlay', () => {
+        rootStore.modules.trade.is_market_closed = true;
+        rootStore.modules.trade.has_only_forward_starting_contracts = false;
```

Comment:
`has_only_forward_starting_contracts` comes from ui store and is `false` by default